### PR TITLE
Feature/improve routing

### PR DESCRIPTION
**Pull request by bdebever**
I have refactored the router so we have a simple loop in the router file. We can create a separate routing configuration file. 

Very flexible as it is an object, so we can add as many attributes as we want. Used that in the past and that's definitely worth it.

### CODE REVIEW

File: frontend/src/react/Router/Router.js [hunk 1]
```diff
@@ -1,15 +1,52 @@
...
-import Instructions from "../Instructions/Instructions";
-import CreateExam from "../Exam/CreateExam";
+
+import Instructions from '../Instructions/Instructions';
+import CreateExam from '../Exam/CreateExam';
+import DisplayExam from '../Exam/DisplayExam';
+import LayoutApp from '../Layout/LayoutApp';
+
+// TODO: move this guy in his own file
+const privateRoutes = {
```
Comment (ValentinChion) [inline frontend/src/react/Router/Router.js:10]: Great work ! Can be used to pass props to routes !

**Review by ValentinChion — APPROVED** [review]

File: frontend/src/react/Router/Router.js [hunk 2]
```diff
@@ -1,15 +1,52 @@
...
+    Instructions: {
+        path: '/newexam/instructions',
+        component: Instructions,
+        title: 'Indiquez les détails du sujet'
+    },
+    CreateExam: {
+        path: '/newexam/exercices',
+        component: CreateExam,
+        title: 'Créer votre examen'
```
Comment (NansD) [inline frontend/src/react/Router/Router.js]: Créez*

File: frontend/src/react/Router/Router.js (hunk 1 above)
Comment (NansD) [reply frontend/src/react/Router/Router.js:10]: So for example we can pass an Exam to `DisplayExam` ? How do we achieve that ?

**Comment by NansD** [comment]
👍

**Review by NansD — APPROVED** [review]

File: frontend/src/react/Router/Router.js (hunk 2 above)
Comment (bdebever) [reply frontend/src/react/Router/Router.js]: Franglish club took over my French! Good catch thanks!

File: frontend/src/react/Router/Router.js (hunk 1 above)
Comment (bdebever) [reply frontend/src/react/Router/Router.js:10]: I don't think we should pass an exam to DisplayExam. We have the id in the route, so we call Redux on the didMount hook in the DisplayExam component. 

If you wanted to pass straight from the Router to DisplayExam, simply add a property to your privateRoutes object, and voila!

What would be the goal of doing that @NansD?

Comment (NansD) [reply frontend/src/react/Router/Router.js:10]: yeah I just wanted to have an example for passing props to routes, as @ValentinChion suggested. Didn't aim to talk about a "real" usecase

**Pull request merged**